test/node/src/join.test.ts

```diff
@@ -778,6 +809,41 @@ for (const dialect of DIALECTS) {
             { first_name: 'Sylvester', name: 'Hammo' },
           ])
         })
+
+        it('should cross join an expression laterally', async () => {
+          const query = ctx.db
+            .selectFrom('person')
+            .crossJoinLateral(
+              (eb) =>
+                eb
+                  .selectFrom('pet')
+                  .innerJoin('person as owner', 'owner.id', 'pet.owner_id')
+                  .select('name')
+                  .whereRef('owner.gender', '=', 'person.gender')
+                  .as('p'),
+            )
+            .select(['first_name', 'p.name'])
+            .orderBy(['first_name', 'p.name'])
+
+          testSql(query, dialect, {
+            postgres: {
+              sql: `select "first_name", "p"."name" from "person" cross join lateral (select "name" from "pet" inner join "person" as "owner" on "owner"."id" = "pet"."owner_id" where "owner"."gender" = "person"."gender") as "p" order by "first_name", "p"."name"`,
+              parameters: [],
+            },
+            mysql: NOT_SUPPORTED,
```

Comment:
should also be supported in MySQL according to [spec](https://dev.mysql.com/doc/refman/8.4/en/join.html).